### ISSUE TITLE
Blocked users' toots still appear on the profiles of users who retoot them

### ISSUE THREAD

**Issue by Skirmisher** [tootsuite/mastodon#251]
Reworded: The posts on the profile page of a user are not filtered for blocked users, so retoots of them still appear. (Yes, I'm going to keep using that word.) This affects the home-interface user view and dedicated profile page view. (As a side note, those feel like different things, but it's hard to refer to them separately--what's the deal there?)

Thanks to @neuroghast@mastodon.social for pointing this out!

**Comment by grimedango** [comment]
Seems like the filtering needs to be applied to those views, yes. This would be great to have implemented (if there isnt anything in particular stopping it from being implemented) cuz its a problem that still plagues other platforms.

**Comment by hoodiek** [comment]
i just confirmed this issue still exists

**Comment by ineffyble** [comment]
There has been some commits since the 16th in this area. Could someone verify if this issue is solved?

**Comment by Gargron** [comment]
Nope, we're still not filtering on that API method atm

**Comment by wxcafe** [comment]
this is still a thing.

**Comment by Cassolotl** [comment]
I just saw and accidentally BOOSTED a toot by someone I block, in someone else's profile. This should absolutely NOT be possible!

**Comment by nynhex** [comment]
Is this still reproducible on master?  I can bootstrap and try to reproduce and submit a PR.

**Comment by Cassolotl** [comment]
@nynhex It's still happening on mastodon.social!

**Comment by trwnh** [comment]
Ins't this the difference between "blocking" and "muting" someone?

Block = "I don't want you to see anything from me."
Mute = "I don't want to see anything from you."

**Comment by Cassolotl** [comment]
@trwnh When you block someone, it *also* mutes them.

**Comment by trwnh** [comment]
@Cassolotl so then, do toots from *muted* users appear when boosted? Because you can block and mute someone separately, and I have some people blocked but not muted. AFAIK I've always had to explicitly do both actions.

**Comment by Cassolotl** [comment]
@trwnh I don't know, but that's okay because this particular issue/bug report is for blocked people, and I shouldn't have to mute AND block to stop seeing someone's toots anywhere.

**Comment by Cassolotl** [comment]
Instance mod of vulpine.club said I could pass [this](https://vulpine.club/@mxsparks/99750775765519966) along:

>if i'm blocked by someone, it seems like i shouldn't be able to see their toots when other people boost them, except that I can

There are a whole bunch of replies from people who have given up on blocks working like they are supposed to. :S

**Comment by nightpool** [comment]
most of those replies sounded like people interacting with pleroma/gnu
social instances. pleroma doesn't support blocks and I don't know how we
can fix that except asking them to support blocks or defederating with them

**Comment by Cassolotl** [comment]
When it happens to me it's always people from other Mastodon instances. Only one replier in that thread mentions Pleroma.

**Comment by nightpool** [comment]
yeah the boosting problem is legit. I'm talking about the overall "I gave
up on things not being visible to people I blocked" sentiment

**Comment by trwnh** [comment]
Late reply / clarification to @Cassolotl : I agree you shouldn't have to mute AND block someone to stop seeing them -- muting them should do that.

If behavior changed such that blocking would ALSO mute someone, that would be a regression and a loss of functionality to me. There are some people who I want to be able to see, but who I also don't want them able to see me -- consider the case of an overzealous family member attempting to scan your social media, who you don't want to hide entirely but simply want them to not be able to see your posts. Although, this might perhaps be served better by a slightly more granular permissions model rather than a simple "hide me from them" / "hide them from me" bifurcation -- out of scope for this issue, though.

**Comment by Cassolotl** [comment]
Blocking someone also removes them from your notifications though. I don't know, I'm not super knowledgeable about this but I feel like if I have blocked someone I shouldn't be seeing them anywhere at all. I don't think I should have to mute them as well.

**Comment by mal0ki** [comment]
I want to work over how blocks and mutes display, for improved UX. I was thinking of bringing it up for discussion on discourse, where we can have more space to just talk, and slide off topic etc. 

I have an idea, that I did put down on paper, and shared in the discord.

**Comment by trwnh** [comment]
Seems there's been a regression at some point: when muting someone and then going to the profile of another person who boosted them, I am now able to see those boosts, where previously I wasn't able to before.

Situation:
1) A boosts B.
2) I mute B from my timeline.
3) I load A's profile.

Expected behavior:
4) A's boost of B should no longer appear for me in web UI.

Actual behavior:
4) A's boost of B is still visible.
5) The context menu option reads "mute user B" as if they aren't already muted.
6) Muting B again will hide them in the web UI, but refreshing the page will bring them back.

If I mute someone then I shouldn't have to see their posts anywhere in timelines or in profiles. Since #7231 was supposed to remove disparities between blocks and mutes, I assume the same would happen were I to block user B.

**Comment by Gargron** [comment]
Fixed this but forgot to x-reference
